crates/runc-shim/src/processes.rs

```diff
@@ -83,7 +87,7 @@ pub struct ProcessTemplate<S> {
 impl<S> ProcessTemplate<S> {
     pub fn new(id: &str, stdio: Stdio, lifecycle: S) -> Self {
         Self {
-            state: Status::CREATED,
+            state: tokio::sync::Mutex::new(Status::CREATED),
```

Comment:
Any reason why we require mutex? Seems like all functions are mutable.

Reply:
I think, the state should be atomic. State changes should be made by only one thread.

Reply:
Correct, so Rust will enforce this at compile time.
Functions which have `&mut self` mutate state, so you won't be able to call them from multiple threads.

Reply:
Got it, thanks for your explanation.

Reply:
Done.